src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -89,6 +97,29 @@ class WorkspaceService(protected val argUserToken: WithAccessToken, val rawlsDAO
     }
   }
 
+  def getCatalog(workspaceNamespace: String, workspaceName: String, userInfo: UserInfo): Future[PerRequestMessage] = {
+    hasAccessOrAdmin(workspaceNamespace, workspaceName, WorkspaceAccessLevels.Read, userInfo) flatMap { hasAccess =>
+      if (hasAccess) {
+        rawlsDAO.getCatalog(workspaceNamespace, workspaceName) map (RequestComplete(_))
+      }
+      else {
+        Future.successful(RequestCompleteWithErrorReport(Forbidden, "must have read+ for the workspace or be an admin to get catalog permissions"))
+      }
+    }
+  }
+
+  def updateCatalog(workspaceNamespace: String, workspaceName: String, updates: Seq[WorkspaceCatalog], userInfo: UserInfo): Future[PerRequestMessage] = {
+    // can update if admin or owner of workspace
+    hasAccessOrAdmin(workspaceNamespace, workspaceName, WorkspaceAccessLevels.Owner, userInfo) flatMap { hasAccess =>
+      if (hasAccess) {
+        rawlsDAO.patchCatalog(workspaceNamespace, workspaceName, updates) map (RequestComplete(_))
+      }
+      else {
+        Future.successful(RequestCompleteWithErrorReport(Forbidden, "must be an owner or admin to set catalog permissions"))
```

Comment:
see comment below about an `asPermitted` wrapper. If you *don't* implement `asPermitted`, this should be `Future.failed` (same for getCatalog).